src/main/java/org/terasology/behaviors/actions/FollowPlayerWithinRangeAction.java

```diff
@@ -64,6 +68,16 @@ public BehaviorState modify(Actor actor, BehaviorState result) {
                 continue;
             }
             if (locationComponent.getWorldPosition().distanceSquared(actorPosition) <= maxDistanceSquared) {
+                if (charactersWithinRange.size() == 0) {
+                    closestCharacter = character;
```

Comment:
Does this mean that if there are no characters in range, the search returns the searching entity?

Reply:
This line indicates that initially closestCharacter is initialized as the first character in the list, and minDistanceFromCharacter is assigned accordingly. After that, since this gets added to the charactersWithinRange list, the size if no longer 0, and then we compare the other characters with this initialised closestCharacter.

Reply:
Looks good, I needed to look at the code in context, not just the snippet :D